openhands/integrations/bitbucket/bitbucket_service.py

```diff
@@ -0,0 +1,418 @@
+                    response = await self.execute_request(
+                        client=client,
+                        url=url,
+                        headers=bitbucket_headers,
+                        params=params,
+                        method=method,
+                    )
+                response.raise_for_status()
+                return response.json(), dict(response.headers)
+        except httpx.HTTPStatusError as e:
+            raise self.handle_http_status_error(e)
+        except httpx.HTTPError as e:
+            raise self.handle_http_error(e)
+
+    async def get_user(self) -> User:
+        """Get the authenticated user's information."""
+        url = f'{self.BASE_URL}/user'
+        data, _ = await self._make_request(url)
+
+        # Convert account_id to an integer if possible, or use a default value
+        account_id = data.get('account_id', '0')
+        try:
+            # Try to extract a numeric part from the account_id string
+            # For example, from "712020:d41205d7-cb10-4142-bdd0-9778094e4710"
+            # we would extract 712020
+            numeric_part = account_id.split(':')[0]
+            id_value = int(numeric_part)
+        except (ValueError, IndexError):
+            # If conversion fails, use a default value
+            id_value = 0
+
+        return User(
+            id=id_value,
+            login=data.get('username', ''),
+            avatar_url=data.get('links', {}).get('avatar', {}).get('href', ''),
+            name=data.get('display_name'),
+            email=None,  # Bitbucket API doesn't return email in this endpoint
+        )
+
+    async def search_repositories(
+        self,
+        query: str,
+        per_page: int,
+        sort: str,
+        order: str,
+    ) -> list[Repository]:
+        """Search for repositories."""
+        # Bitbucket doesn't have a dedicated search endpoint like GitHub
+        # We'll use the repositories endpoint and filter client-side
+        url = f'{self.BASE_URL}/repositories'
+        params = {
+            'q': query,
+            'pagelen': per_page,
+            'sort': sort,
+        }
+
+        data, headers = await self._make_request(url, params)
+
+        repositories = []
+        for repo in data.get('values', []):
+            repositories.append(
+                Repository(
+                    id=repo.get('uuid', ''),
+                    full_name=f'{repo.get("workspace", {}).get("slug", "")}/{repo.get("slug", "")}',
+                    git_provider=ProviderType.BITBUCKET,
+                    is_public=repo.get('is_private', True) is False,
+                    stargazers_count=None,  # Bitbucket doesn't have stars
+                    link_header=headers.get('Link', ''),
+                    pushed_at=repo.get('updated_on'),
+                )
+            )
+
+        return repositories
+
+    async def get_repositories(self, sort: str, app_mode: AppMode) -> list[Repository]:
```

Comment:
@openhands for `get_repositories` query, `{self.BASE_URL}/repositories` is for the user's repos which are public

using `{self.BASE_URL}/workspaces` to get workspaces and the repos inside it is for the user's private repos 

please split this method into two private methods; then aggregate their results here and return it